google-cloud-bigtable/src/test/java/com/google/cloud/bigtable/admin/v2/it/BigtableCmekIT.java

```diff
@@ -0,0 +1,147 @@
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Instant;
+import org.threeten.bp.temporal.ChronoUnit;
+
+@RunWith(JUnit4.class)
+public class BigtableCmekIT {
+  private static final String CMEK_KMS_KEY_PROPERTY_NAME = "bigtable.kms_key_name";
+
+  @ClassRule public static TestEnvRule testEnvRule = new TestEnvRule();
+
+  private static String kmsKeyName;
+  private String instanceId;
+  private String clusterId;
+
+  private BigtableInstanceAdminClient instanceAdmin;
+  private BigtableTableAdminClient tableAdmin;
+
+  @BeforeClass
+  public static void validatePlatform() {
+    assume()
+        .withMessage("Emulator doesn't support on CMEK")
+        .that(testEnvRule.env())
+        .isNotInstanceOf(EmulatorEnv.class);
+
+    kmsKeyName = System.getProperty(CMEK_KMS_KEY_PROPERTY_NAME);
+
+    assume().withMessage("%s property is not set, skipping CMEK test", kmsKeyName);
+  }
+
+  @Before
+  public void setUp() throws IOException {
+    instanceId = AbstractTestEnv.TEST_INSTANCE_PREFIX + Instant.now().getEpochSecond();
+    clusterId = instanceId + "-c1";
+
+    instanceAdmin = testEnvRule.env().getInstanceAdminClient();
+    tableAdmin =
+        BigtableTableAdminClient.create(
+            testEnvRule
+                .env()
+                .getTableAdminSettings()
+                .toBuilder()
+                .setInstanceId(instanceId)
+                .build());
+  }
+
+  @After
+  public void teardown() {
+    try {
+      for (String backup : tableAdmin.listBackups("-")) {
+        tableAdmin.deleteBackup(clusterId, backup);
+      }
+    } catch (NotFoundException ignored) {
+    }
+
+    try {
+      instanceAdmin.deleteInstance(instanceId);
+    } catch (NotFoundException ignored) {
+    }
+
+    tableAdmin.close();
+  }
+
+  @Test
+  public void test() throws Exception {
+    String newTableId = "new-table";
+    String backupId = "my-backup";
+
+    String zoneId = testEnvRule.env().getPrimaryZone();
+
+    instanceAdmin.createInstance(
+        CreateInstanceRequest.of(instanceId)
+            .addCmekCluster(clusterId, zoneId, 1, StorageType.SSD, kmsKeyName));
+
+    tableAdmin.createTable(CreateTableRequest.of(newTableId).addFamily("cf"));
+
+    tableAdmin.createBackup(
+        CreateBackupRequest.of(clusterId, backupId)
+            .setExpireTime(Instant.now().plus(6, ChronoUnit.HOURS))
+            .setSourceTableId(newTableId));
+
+    Map<String, List<EncryptionInfo>> encryptionInfos = tableAdmin.getEncryptionInfo(newTableId);
+    assertThat(encryptionInfos).hasSize(1);
+    assertThat(encryptionInfos.get(clusterId)).hasSize(1);
+
+    EncryptionInfo encryptionInfo = encryptionInfos.get(clusterId).get(0);
+
+    assertThat(encryptionInfo.getType()).isEqualTo(EncryptionInfo.Type.CUSTOMER_MANAGED_ENCRYPTION);
+    assertThat(encryptionInfo.getStatus().getCode()).isAnyOf(Status.Code.OK, Status.Code.UNKNOWN);
+    if (encryptionInfo.getStatus().getCode() == Code.UNKNOWN) {
```

Comment:
Can you add a comment explaining why this is nondeterministic? Or better yet, can it be eliminated? ie poll while UNKNOWN and assert that final state is OK?

Reply:
Yea I had added comments but move them closer to the code. I basically added a new system property that testers can set to true to poll for final state.